### PR TITLE
[RFC][SelectionDAG] Add and use SDNode::getAsZExtVal() helper

### PR DESCRIPTION
**Pull request by asb**
This follows on from #76708, allowing
`cast<ConstantSDNode>(N)->getZExtValue()` to be replaced with just `N->getAsZextVal();`
    
Introduced via `git grep -l "cast<ConstantSDNode>\(.*\).*getZExtValue" |
xargs sed -E -i 's/cast<ConstantSDNode>\((.*)\)->getZExtValue/\1->getAsZExtVal/'` and then using `git clang-format` on the result.

To me it seems this helper is filling in a logical gap given that the getConstantOperandVal helper already exists, but I'd very much welcome alternate viewpoints.

### CODE REVIEW

File: llvm/include/llvm/CodeGen/SelectionDAGNodes.h [hunk 1]
```diff
@@ -923,6 +923,7 @@ END_TWO_BYTE_PACK()
 
   /// Helper method returns the integer value of a ConstantSDNode operand.
   inline uint64_t getConstantOperandVal(unsigned Num) const;
+  inline uint64_t getAsConstantVal() const;
```
Comment (arsenm) [inline llvm/include/llvm/CodeGen/SelectionDAGNodes.h:926]: Potentially confusing if you need a signed immediate, but that's already an issue with the existing helper

**Review by arsenm — APPROVED** [review]

File: llvm/include/llvm/CodeGen/SelectionDAGNodes.h (hunk 1 above)
Comment (asb) [reply llvm/include/llvm/CodeGen/SelectionDAGNodes.h:926]: Yes, I'm definitely inheriting that confusion. Would getAsZExtVal be better?

Comment (asb) [reply llvm/include/llvm/CodeGen/SelectionDAGNodes.h]: I've gone ahead and renamed to getAsZExtVal instead and rebased now that #76708 landed.

**Review by RKSimon — APPROVED** [review]
LGTM @arsenm Are you happy with the name change?

**Review by arsenm — APPROVED** [review]

**Pull request merged**